lib/elixir/lib/calendar/date.ex

```diff
@@ -659,6 +659,106 @@ defmodule Date do
     calendar.day_of_week(year, month, day)
   end
 
+  @doc """
+  Calculates the day of the year of a given `date`.
+
+  Returns the day of the year as an integer. For the ISO 8601
+  calendar (the default), it is an integer from 1 to 366.
+
+  ## Examples
+
+      iex> Date.day_of_year(~D[2016-01-01])
+      1
+      iex> Date.day_of_year(~D[2016-11-01])
+      306
+      iex> Date.day_of_year(~D[-0015-10-30])
+      303
+      iex> Date.day_of_year(~D[2004-12-31])
+      366
+
+  """
+  @doc since: "1.8.0"
+  @spec day_of_year(Calendar.date()) :: non_neg_integer()
```

Comment:
Maybe?

```suggestion
  @spec day_of_year(Calendar.date()) :: Calendar.day()
```

Reply:
Ooops! Nop. But maybe `pos_integer`, or this could return 0?

Reply:
I'm not aware of any calendars that are 0-based for days (I'll check again) so I think `pos_integer` is probably fine.  `Calendar.day` is defined as `integer`.  I wonder if that should be `pos_integer`?

Reply:
@kipcole9 I think so, yes. :+1: